### PR TITLE
Implement position and monster moving

### PR DESCRIPTION
**Pull request by IHIHHIHI**
This revision includes:
- Add component 'Position'
- Add component 'Distance'
- Add system to update monster's path

### CODE REVIEW

File: Includes/CubbyTower/Components/Distance.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+namespace CubbyTower
+{
+//!
+//! \brief Distance struct.
+//!
+//! This struct stores the distance a monster has traveled.
+//!
+struct Distance
+{
+    double s;
```
Comment (utilForever) [inline Includes/CubbyTower/Components/Distance.hpp]:
```suggestion
    double distance;
```

File: Includes/CubbyTower/Systems/Path.hpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// property of any third parties.
+
+#ifndef CUBBYTOWER_PATH_HPP
+#define CUBBYTOWER_PATH_HPP
+
+#include <entt/entt.hpp>
+
+namespace CubbyTower
+{
+
```
Comment (utilForever) [inline Includes/CubbyTower/Systems/Path.hpp]:
```suggestion
```

File: Includes/CubbyTower/Systems/Path.hpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+//! This function called every frame and gives enemies' position using distance
```
Comment (utilForever) [inline Includes/CubbyTower/Systems/Path.hpp]:
```suggestion
//! This function called every frame and gives the position of enemies using distance.
```

File: Includes/CubbyTower/Systems/Path.hpp [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+//! \param registry A registry that handles entities.
+void Path(entt::registry& registry);
```
Comment (utilForever) [inline Includes/CubbyTower/Systems/Path.hpp]:
```suggestion
void UpdatePathSystem(entt::registry& registry);
```

File: Sources/CubbyTower/Systems/Path.cpp [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+// property of any third parties.
+
+#include <CubbyTower/Commons/Tags.hpp>
+#include <CubbyTower/Components/Position.hpp>
+#include <CubbyTower/Components/Distance.hpp>
+#include <CubbyTower/Systems/Path.hpp>
+
+namespace CubbyTower
+{
+void Path(entt::registry& registry)
```
Comment (utilForever) [inline Sources/CubbyTower/Systems/Path.cpp]:
```suggestion
void UpdatePathSystem(entt::registry& registry)
```

File: Sources/CubbyTower/Systems/Path.cpp [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+{
+    auto view = registry.view<Distance,Position>();
```
Comment (utilForever) [inline Sources/CubbyTower/Systems/Path.cpp]:
```suggestion
    auto view = registry.view<Distance, Position>(); 
```

File: Sources/CubbyTower/Systems/Path.cpp [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace CubbyTower
+{
+void Path(entt::registry& registry)
+{
+    auto view = registry.view<Distance,Position>(); 
+    for (auto enemy : view)
+    {
+        auto &pos = view.get<Position>(enemy);
+        auto &dis = view.get<Distance>(enemy);
```
Comment (utilForever) [inline Sources/CubbyTower/Systems/Path.cpp]:
```suggestion
        auto& pos = view.get<Position>(enemy);
        auto& dist = view.get<Distance>(enemy);
```

File: Sources/CubbyTower/Systems/Path.cpp [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        pos.x = dis.s;
```
Comment (utilForever) [inline Sources/CubbyTower/Systems/Path.cpp]:
```suggestion
        pos.x = dist.distance;
```

File: Includes/CubbyTower/Systems/Path.hpp [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+// We are making my contributions/submissions to this project solely in our
+// personal capacity and are not conveying any rights to any intellectual
+// property of any third parties.
+
+#ifndef CUBBYTOWER_PATH_HPP
+#define CUBBYTOWER_PATH_HPP
+
+#include <entt/entt.hpp>
+
+namespace CubbyTower
```
Comment (utilForever) [inline Includes/CubbyTower/Systems/Path.hpp]:
```suggestion
namespace CubbyTower::Path
```

File: Sources/CubbyTower/Systems/Path.cpp [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+// We are making my contributions/submissions to this project solely in our
+// personal capacity and are not conveying any rights to any intellectual
+// property of any third parties.
+
+#include <CubbyTower/Commons/Tags.hpp>
+#include <CubbyTower/Components/Position.hpp>
+#include <CubbyTower/Components/Distance.hpp>
+#include <CubbyTower/Systems/Path.hpp>
+
+namespace CubbyTower
```
Comment (utilForever) [inline Sources/CubbyTower/Systems/Path.cpp]:
```suggestion
namespace CubbyTower::Path
```

File: Tests/UnitTests/Systems/PathTests.cpp [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+{
+    //Testing if position is changing as distance changes
+
+    entt::registry registry;
+
+    auto enemy1 = registry.create();
+    registry.emplace<Tag::Enemy>(enemy1);
+    registry.emplace<Position>(enemy1 , 0.0, 0.0);
+    registry.emplace<Distance>(enemy1 , 5.0);
+    Path(registry);
```
Comment (utilForever) [inline Tests/UnitTests/Systems/PathTests.cpp]:
```suggestion
    Path::UpdatePathSystem(registry);
```

File: Tests/UnitTests/Systems/PathTests.cpp [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+TEST_CASE("[Path] - Path")
+{
+    //Testing if position is changing as distance changes
+
+    entt::registry registry;
+
+    auto enemy1 = registry.create();
+    registry.emplace<Tag::Enemy>(enemy1);
+    registry.emplace<Position>(enemy1 , 0.0, 0.0);
+    registry.emplace<Distance>(enemy1 , 5.0);
```
Comment (utilForever) [inline Tests/UnitTests/Systems/PathTests.cpp]:
```suggestion
    registry.emplace<Position>(enemy1, 0.0, 0.0);
    registry.emplace<Distance>(enemy1, 5.0);
```

File: Tests/UnitTests/Systems/PathTests.cpp [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    Path(registry);
+    for (auto [enemy, pos] : registry.view<Tag::Enemy , Position>().each())
```
Comment (utilForever) [inline Tests/UnitTests/Systems/PathTests.cpp]:
```suggestion
    for (auto [enemy, pos] : registry.view<Tag::Enemy, Position>().each())
```

File: Tests/UnitTests/Systems/PathTests.cpp [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+#include <CubbyTower/Systems/Path.hpp>
+
+#include <entt/entt.hpp>
+
+using namespace CubbyTower;
+
+TEST_CASE("[Path] - Path")
+{
+    //Testing if position is changing as distance changes
+
```
Comment (utilForever) [inline Tests/UnitTests/Systems/PathTests.cpp]: Delete these lines.
```suggestion
```

File: Tests/UnitTests/Systems/PathTests.cpp [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+#include <CubbyTower/Commons/Tags.hpp>
+#include <CubbyTower/Components/Position.hpp>
+#include <CubbyTower/Components/Distance.hpp>
+#include <CubbyTower/Systems/Path.hpp>
+
+#include <entt/entt.hpp>
+
+using namespace CubbyTower;
+
+TEST_CASE("[Path] - Path")
```
Comment (utilForever) [inline Tests/UnitTests/Systems/PathTests.cpp]:
```suggestion
TEST_CASE("[PathSystem] - UpdatePathSystem")
```

File: Tests/UnitTests/Systems/PathTests.cpp [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+// We are making my contributions/submissions to this project solely in our
+// personal capacity and are not conveying any rights to any intellectual
+// property of any third parties.
+
+#include <doctest.h>
+
+#include <CubbyTower/Commons/Tags.hpp>
+#include <CubbyTower/Components/Position.hpp>
+#include <CubbyTower/Components/Distance.hpp>
+#include <CubbyTower/Systems/Path.hpp>
```
Comment (utilForever) [inline Tests/UnitTests/Systems/PathTests.cpp]:
```suggestion
#include <CubbyTower/Systems/PathSystem.hpp>
```

File: Sources/CubbyTower/Systems/Path.cpp [hunk 17]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2021 CubbyTower Team
+// Chris Ohk, Minkyu Lee, Minjune Yi
+// We are making my contributions/submissions to this project solely in our
+// personal capacity and are not conveying any rights to any intellectual
+// property of any third parties.
+
+#include <CubbyTower/Commons/Tags.hpp>
+#include <CubbyTower/Components/Position.hpp>
+#include <CubbyTower/Components/Distance.hpp>
+#include <CubbyTower/Systems/Path.hpp>
```
Comment (utilForever) [inline Sources/CubbyTower/Systems/Path.cpp]:
```suggestion
#include <CubbyTower/Systems/PathSystem.hpp>
```

**Review by utilForever — CHANGES_REQUESTED** [review]
You should check my feedback.
Also, I suggest change the file name 'Path.hpp/cpp' to 'PathSystem.hpp/cpp' and 'PathTests.cpp' to 'PathSystemTests.cpp'.

File: Includes/CubbyTower/Systems/Path.hpp (hunk 2 above)
Comment (IHIHHIHI) [reply Includes/CubbyTower/Systems/Path.hpp]: 14 번 줄 메모

**Review by utilForever — APPROVED** [review]
LGTM now. Thanks for your work! 🎉

**Review by red1108 — APPROVED** [review]
LGTM.
I wonder why you add ::Path

**Comment by utilForever** [comment]
@red1108 To improve readability for code. 👓

**Pull request merged**